src/temporal_logic/rules.rs

```diff
@@ -144,7 +144,7 @@ proof fn implies_apply_with_always<T>(ex: Execution<T>, p: TempPred<T>, q: TempP
     always_unfold::<T>(ex, p);
 }
 
-proof fn entails_apply<T>(ex: Execution<T>, p: TempPred<T>, q: TempPred<T>)
+pub proof fn entails_apply<T>(ex: Execution<T>, p: TempPred<T>, q: TempPred<T>)
```

Comment:
I guess this one does not have to be `pub` if you simplify `lemma_always_forall_resp_matches_at_most_one_pending_req` as mentioned above?